src/index.ts

```diff
@@ -1,173 +1,242 @@
-    }
-  };
+  // append a default stylesheet for tooltips to the head
+  if (!options.disableDefaultStyle && !document.getElementById(STYLE_ID)) {
+    const style = document.createElement('style');
+    style.setAttribute('id', STYLE_ID);
+    style.innerHTML = STYLE;
+    document.querySelector('head')!.appendChild(style);
+  }
+
+  tooltipElement = document.createElement('div');
+  tooltipElement.setAttribute('id', options.id);
+  tooltipElement.classList.add('vg-tooltip');
+
+  document.querySelector('body')!.appendChild(tooltipElement);
 }
 
-export function vegaLite(vgView: VgView, vlSpec: TopLevelSpec, options: Option = {showAllFields: true, isComposition: false}) {
-  options = supplementOptions(copyOptions(options), vlSpec);
-  start(vgView, copyOptions(options));
+/**
+ * Format the value to be shown in the toolip.
+ *
+ * @param value The value to show in the tooltip.
+ */
+function formatValue(value: any): string {
+  if (isArray(value)) {
+    return `[${value.map(v => (isString(v) ? v : stringify(v))).join(', ')}]`;
+  }
 
-  return {
-    destroy: function () {
-      // remove event listeners
-      vgView.removeEventListener('mouseover.tooltipInit');
-      vgView.removeEventListener('mousemove.tooltipUpdate');
-      vgView.removeEventListener('mouseout.tooltipRemove');
+  if (isString(value)) {
+    return value;
+  }
 
-      cancelPromise(); // clear tooltip promise
-    }
-  };
-}
+  if (isObject(value)) {
+    let content = '';
 
-function start(vgView: VgView, options: Option) {
-  // initialize tooltip with item data and options on mouse over
-  vgView.addEventListener('mouseover.tooltipInit', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      // clear existing promise because mouse can only point at one thing at a time
-      cancelPromise();
-
-      // make a new promise with time delay for tooltip
-      tooltipPromise = window.setTimeout(function () {
-        init(event, item, options);
-      }, options.delay || DELAY);
-    }
-  });
+    const { title, ...rest } = value as any;
 
-  // update tooltip position on mouse move
-  // (important for large marks e.g. bars)
-  vgView.addEventListener('mousemove.tooltipUpdate', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item) && tooltipActive) {
-      update(event, item, options);
+    if (title) {
+      content += `<h2>${title}</h2>`;
     }
-  });
-
-  // clear tooltip on mouse out
-  vgView.addEventListener('mouseout.tooltipRemove', function (event: MouseEvent, item: Scenegraph) {
-    if (shouldShowTooltip(item)) {
-      cancelPromise();
 
-      if (tooltipActive) {
-        clear(event, item, options);
+    content += '<table>';
+    for (const key of Object.keys(rest)) {
+      let val = (rest as any)[key];
+      if (isObject(val)) {
+        val = stringify(val);
       }
+
+      content += `<tr><td class="key">${escape(key)}:</td><td class="value">${escape(val)}</td></tr>`;
```

Comment:
escape() here is both deprecated and incorrect.  You want to escape just the `<` and `&` symbols with corresponding values.  See https://stackoverflow.com/questions/6234773/can-i-escape-html-special-chars-in-javascript -- but you don't need to escape anything else because its not part of the attribute.